x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/classification/Accuracy.java

```diff
@@ -0,0 +1,165 @@
+import org.elasticsearch.xpack.core.ml.dataframe.evaluation.EvaluationMetricResult;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * {@link Accuracy} is a metric that answers the question:
+ *   "What fraction of examples have been classified correctly by the classifier?"
+ *
+ * equation: accuracy = 1/n * Σ(y == y´)
+ */
+public class Accuracy implements ClassificationMetric {
+
+    public static final ParseField NAME = new ParseField("accuracy");
+
+    private static final String PAINLESS_TEMPLATE = "doc[''{0}''].value == doc[''{1}''].value";
+    private static final String AGG_NAME = "classification_" + NAME.getPreferredName();
+
+    private static String buildScript(Object...args) {
+        return new MessageFormat(PAINLESS_TEMPLATE, Locale.ROOT).format(args);
+    }
+
+    private static final ObjectParser<Accuracy, Void> PARSER = new ObjectParser<>("accuracy", true, Accuracy::new);
+
+    public static Accuracy fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    private EvaluationMetricResult result;
+
+    public Accuracy() {}
+
+    public Accuracy(StreamInput in) throws IOException {}
+
+    @Override
+    public String getWriteableName() {
+        return NAME.getPreferredName();
+    }
+
+    @Override
+    public String getName() {
+        return NAME.getPreferredName();
+    }
+
+    @Override
+    public final List<AggregationBuilder> aggs(String actualField, String predictedField) {
+        if (result != null) {
+            return List.of();
+        }
+        return List.of(AggregationBuilders.avg(AGG_NAME).script(new Script(buildScript(actualField, predictedField))));
+    }
+
+    @Override
+    public void process(Aggregations aggs) {
+        if (result == null && aggs.get(AGG_NAME) != null) {
+            NumericMetricsAggregation.SingleValue value = aggs.get(AGG_NAME);
+            result = value == null ? new Result(0.0) : new Result(value.value());
+        }
+    }
+
+    @Override
+    public Optional<EvaluationMetricResult> getResult() {
+        return Optional.ofNullable(result);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(NAME.getPreferredName());
+    }
+
+    public static class Result implements EvaluationMetricResult {
+
+        private static final String ACCURACY = "accuracy";
```

Comment:
I think we will want to include individual label accuracy as well. 

What about changing this to `overall_accuracy` to avoid future confusion when we add individual labels accuracy?

Reply:
Done.